Add scaling factor for webvr renderers

joshpatt2 opened the pull request:
The problem is some browser send smaller width/heights then resolution (Such as Oculus Browser), causing image to not be as crisp as possible.

This is a fix for #14113

cvan commented on src/renderers/webvr/WebVRManager.js:
```diff
@@ -59,8 +60,8 @@ function WebVRManager( renderer ) {
 			var eyeParameters = device.getEyeParameters( 'left' );
-			var renderWidth = eyeParameters.renderWidth;
-			var renderHeight = eyeParameters.renderHeight;
+			var renderWidth = eyeParameters.renderWidth * renderScaling;
```
thanks for adding this. it helps :+1:

joshpatt2 replied:
No problem, thanks for the review! 👍

joshpatt2 commented on src/renderers/webvr/WebVRManager.js:
```diff
@@ -17,6 +17,7 @@ function WebVRManager( renderer ) {
 	var frameData = null;
 
 	var poseTarget = null;
+	var renderScaling = 1.0;
```
@cvan ah yeah, that makes a lot of sense! I'm away on holidays for 8 days, will update it when I get back :)

joshpatt2 commented:
@cvan - I've updated the change with your feedback, can you take another look when you have a chance, thanks! :)

cvan commented on src/renderers/webvr/WebVRManager.js:
```diff
@@ -59,8 +59,8 @@ function WebVRManager( renderer ) {
 			var eyeParameters = device.getEyeParameters( 'left' );
-			var renderWidth = eyeParameters.renderWidth;
-			var renderHeight = eyeParameters.renderHeight;
+			var renderWidth = eyeParameters.renderWidth * scope.renderScaling;
```
should `scope.renderScaling` not be used when a VR device is not presenting but the WebVR renderer is `enabled`?

joshpatt2 replied:
@cvan - hmm...I'm not sure that it should be.  In the case that we're not presenting we use the set size of the renderer, which the user can set.  The problem when we're presenting is the eye parameters may be smaller then the actual resolution, so we need to try to scale it if we want 1:1 pixel mapping.  I'm happy to update it for the non presenting case though if you think it would be good to be consistent here.

joshpatt2 replied:
@cvan - following up on this, what do you think? Should I modify the else clause as well or just use the canvas element size as set?

joshpatt2 replied:
@cvan what action do you want taken here? I'm happy to make the change you suggest but want to make sure you're ok with the side effect, i.e. it looks like in the non presenting case we just respect the canvas width/height that was specified, should we restore this with a vr scaling factor applied to it? That feels a bit strange.

mrdoob commented:
How would this interface with the WebXR API?

joshpatt2 commented:
@mrdoob funny you should mention that...looks like it just breaks now since the api isn't on the WebXRManager, I'll fix that.  I think it should use framebufferScaleFactor but i need to learn a bit more about that.

joshpatt2 commented:
@mrdoob i've updated it so the WebVR part better aligns to the webxr part...haven't been able to test it through, i'm having issues with chrome and webxr on my test page...

Commit:
Add scaling factor for webvr render size

  Oculus browser is reporting a smaller width/height than actual resolution.  Because of this we lose sharpness.  I'm adding a scaling factor so users can adjust it.

joshpatt2 commented:
Tested with webxr to make sure nothing broke with the change:
https://jsfiddle.net/7jc6yboL/

joshpatt2 commented:
@mrdoob anything else needed to land this pull request?

mrdoob commented:
No. I think this is good to go.

mrdoob commented:
Thanks!
